### PR TITLE
Bootstrap javascript components initiator

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue #11654
### Summary of Changes
- No more inline scipts for bootstrap components
- Make Joomla's bootstrap (js) dependancy less strict, as the init script together with the jlayouts dictate the js framework. For example someone can override bootstrap.js, bootstrap-init.js and the layouts and use Foundation or Ukit or any other framework!
#### You will Need to use 3.7 branch for testing!!!

For modals, tooltips, popover and tabs apply this patch and navigate to isis template on the backend. Ensure that no javascript errors are logged and everything still works.
To test the other bootstrap components download this module https://github.com/dgt41/bootstrap-test/raw/gh-pages/bootstrap_test.zip install, publish in the front end and then visit that page (template must be protostar).
Enjoy!
### B/C

There are few layouts that are touched, but joomla's b/c promise states that these are not B/C breaks. Down to PLT for a decision on that.
### Documentation Changes Required

None.

---
This change is [Reviewable](https://reviewable.io/reviews/joomla/joomla-cms/11922)

### CODE REVIEW

**Comment by brianteeman** [comment]
If you are removing a file please add it to the list in script.php

**Comment by dgt41** [comment]
@brianteeman done, thanks!

**Comment by andrepereiradasilva** [comment]
this seems to work fine.

side note: have the other tabs issue (Permissions Tab selected) (https://github.com/joomla/joomla-cms/issues/11648) but not related to this PR.

**Comment by zero-24** [comment]
I have notices that in the backend the `Edit` Tooltips (e.g. com_content) don't work anymore as bs tooltips there are just browser tooltips.

Also I'm not sure if this is a bug here or in the mod if i show it in the frontend it looks like this:
image

See the kitten on the debug console. Also the page is jumping.

What is `Affix` supposed to do?

image

**Comment by dgt41** [comment]
@zero-24 try to use the module in an article with {loadmodule...} or {loadposition...}. This module wasn't created or tested for narrow positions (left, right etc), so that might be a problem.
Affix just sticks the image to a given value (in px) so when you scroll the image should remain in the same position

**Comment by zero-24** [comment]
I have tested this item :white_check_mark: successfully on cbcb9408a0e17249b13bc136e124f1250b129200<br /><br />:+1: Thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11922">issues.joomla.org/tracker/joomla-cms/11922</a>.</sub>

**Comment by hans2103** [comment]
I have tested this item :white_check_mark: successfully on cbcb9408a0e17249b13bc136e124f1250b129200<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11922">issues.joomla.org/tracker/joomla-cms/11922</a>.</sub>

**Comment by dgt41** [comment]
Conflicts resolved

**Comment by roland-d** [comment]
@zero-24 and @hans2103 Can you test once more? Thanks.

**Comment by hans2103** [comment]
I have tested this item :white_check_mark: successfully on 9e0f4bd5dce6361b39a784789ff5a4efb0bad913<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11922">issues.joomla.org/tracker/joomla-cms/11922</a>.</sub>

File: media/system/js/bootstrap-init.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+		/** Dropdown menu **/
+		if (dropdown) {
+			$.each(dropdown, function(index, value) {
+				$('#' + index).dropdown();
+			});
+		}
+
+		/** Modals **/
+		if (modal) {
```
Comment (Fedik) [inline media/system/js/bootstrap-init.js]: it is always will be `true` as on top you use `modal = $('.joomla-modal')`, 
jQuery always return "jQuery object" even if `.joomla-modal` do not exist on the page, 

maybe you meant `if(modal.length)`?

Comment (dgt41) [inline media/system/js/bootstrap-init.js]: Ahhh, I though `modal = Joomla.getOptions('bootstrap.modal')` which of course is not the case....

**Comment by roland-d** [comment]
@dgt41 I am afraid there is a new conflict.

**Comment by dgt41** [comment]
@roland-d and fixed

**Comment by zero-24** [comment]
@dgt41 did you want to close this? So this should be removed from the mileston too?

**Comment by dgt41** [comment]
First of all a big thank you to all the people spend some time with this PR.
An now the explanation why I closed it:
My main idea here was
- remove the inline script
- get the code ready for some `Joomla.ui` API that would freed joomla from any javascript framework.

The later was not accepted from PLT, (public discussion on twitter) so there is no real gain if we just removing inline scripts.

I am sorry that I have wasted people's precious time here 😕

**Comment by zero-24** [comment]
Thanks :+1:

**Pull request closed without merging**